README.md
INTRODUCTION
============

This App provides CAS authentication support, using the phpCAS library of jasig/apereo.

INSTALLATION
============

1\. DEPENDENCIES
---------------

* ownCloud >= 10.0.0 and Nextcloud >= 14.0.0
* PHP >= 5.6, PHP 7.0 if possible
* Optional: [Composer Dependency Manager](https://getcomposer.org/), if you want to install via GIT.
* Optional: your own phpCAS installation with at least phpCAS 1.3.5.

This app does not require a standalone version of jasig’s/apereo’s phpCAS any longer. The library is shipped within composer dependencies, in the archive file you downloaded or the Market/App-Store version if used. Although you can configure to use your own version of jasig’s/apereo’s phpCAS library later on.

2\. Recommended - ownCloud Market:
----------------------------

1. Access the ownCloud web interface with a locally created ownCloud user with admin privileges.
2. Navigate to the market in your ownCloud instance.
3. Navigate to the Security category and find **CAS user and group backend**.
4. Install the app.
5. Access the administrations panel => Apps and enable the **CAS user and group backend** app.
6. Access the administration panel => User Authentication and configure the app.


3\. Basic - Release archive/Nextcloud Appstore:
---------------------------

1. Download the current stable release from [the github releases page](https://github.com/felixrupp/user_cas/releases) according to your platform (ownCloud or Nextcloud) or use the link provided on [apps.nextcloud.com](https://apps.nextcloud.com/apps/user_cas) for Nextcloud.
2. Unzip/Untar the archive.
3. Rename the unarchived folder to `user_cas` if not already named like that.
4. Move the `user_cas` folder to the apps folder of your platform installation.
5. Adjust the settings for the `user_cas` folder according to your webserver setup.
6. Access the platform web interface with a locally created platform user with admin privileges.
7. Access the administrations panel => Apps and enable the **CAS user and group backend** app.
8. Access the administration panel => Security and configure the app.


4\. Advanced for development purposes only – GIT clone with composer:
-------------------------

1. Git clone/copy the downloaded `user_cas` folder into the platform’s apps folder and make sure to set correct permissions for your webserver.
2. Change directory inside `user_cas` folder after cloning and perform a `composer update` command if you installed via GIT. The dependencies will be installed. Attention: You will need the [composer](https://getcomposer.org/doc/00-intro.md#installation-linux-unix-osx) binary to be installed.
3. Adjust the settings for the `user_cas` folder according to your webserver setup.
4. Access the platform web interface with a locally created platform user with admin privileges.
5. Access the administrations panel => Apps and enable the **CAS user and group backend** app.
6. Access the administration panel => User Authentication/Security and configure the app.


**Be aware that you may need to add the following entry to your config/config.php to prevent warning messages about the missing app signature in your backend:**

```
'integrity.ignore.missing.app.signature' =>
    array (
        0 => 'user_cas',
    ),
```


CONFIGURATION
=============

The app is configured by using the administration panel. Please make sure to configure with an admin user, authenticated locally against your ownCloud instance (and not against CAS). Make sure to fill in all the fields provided.


CAS Server
----------

**CAS Server Version**: Default is CAS version 3.0, if you have no special configuration leave it as is.

**CAS Server Hostname**: The host name of the webserver hosting your CAS, lookup /etc/hosts or your DNS configuration or ask your IT-Department.

**CAS Server Port**: The port CAS is listening to. Default for HTTPS should be `443`.

**CAS Server Path**: The directory of your CAS. In common setups this path is `/cas`. Use `/` if your CAS is in your document root.

**Service URL**: Service URL pointing to your plattform used for CAS authentication and redirection. Useful when behind a reverse proxy. This url must end in `/apps/user_cas/login`.

**Certification file path (.crt)**: If you don't want to validate the certificate (i.e. self-signed certificates) then leave this blank. Otherwise enter the absolute path to the certificate (.crt) on your server, beginning at root level.

**Use CAS proxy initialization**: If active, the CAS-Client is initialized as a proxy. Default off. Activate only, if you know what you’re doing.


Basic
-----

**Force user login using CAS?**: If checked, users will immediately be redirected to CAS login page, after visiting the ownCloud URL. If checked, *Disable CAS logout* is automatically disabled. Default: off

**Don’t use force login on these client-IPs**: Comma separated list of client IP addresses (or address ranges), which won’t be forced to login if "Force user login" is enabled (e.g. 192.168.1.1-254,192.168.2.5). Default: empty

**Disable CAS logout**: If checked, you will only be logged out from ownCloud/Nextcloud and not from your CAS instance. Default: off

**Autocreate user after first CAS login?**: If checked, users authenticated against CAS are automatically created. This means, users which did not exist in the database yet who authenticate against CAS will be created and stored in the ownCloud database on their first login. Default: on

**Update user data after each CAS login?**: If checked, the data provided by CAS is used to update ownCloud user attributes each time the user logs in. Default: off

**Disable CAS SingleSignout**: If checked, SingleSignout requests from your CAS server will be ignored. ownCloud/Nextcloud sessions will not be terminated because of SSO. Default: off

**SingleSignout Servers**: Provide a list of servers which can send SingleSignout requests for your CAS ticket (leave empty if you do not have to restrict logout to defined servers).

**Keep CAS-ticket-ids in URL?**: If checked, CAS-ticket-ids are not removed from the URL. Beware: Potential security risk! Only activate, if you know what you are doing. Default off

**Overwrite Login Button Label**: Overwrites the CAS-Login button label (only used in Nextcloud). Default: empty


<a name="mapping"></a>

Mapping
-------

If CAS provides extra attributes, `user_cas` can retrieve the values of them. Since their name differs in various setups it is necessary to map ownCloud-attribute-names to CAS-attribute-names.

**Email**: Name of email attribute in CAS. Default: empty

**Display Name**: Name of display name attribute(s) in CAS (this might be the "real name" of a user or a combination of two fields like: firstnames+surnames). Default: empty

**Group**: Name of group attribute in CAS. Default: empty

**Quota**: Name of quota attribute in CAS. Quota can be a numeric byte value or a human readable string, like 1GB or 512MB. Default: empty


<a name="groups"></a>

Groups
------

**Locked Groups**: Groups that will not be unlinked. These groups are preserved, when updating a user after login and are not unlinked. Please provide a comma separated list without blanks (eg.: group1,group2). Default: empty

**Default group (when autocreating users)**: When auto creating users after authentication, these groups are set as default if the user has no CAS groups. Please provide a comma separated list without blanks (eg.: group1,group2). Default: empty

**Authorized CAS Groups**: Members of these groups are authorized to use the ownCloud instance. This setting is especially helpful, if your CAS instance is not handling authorization itself. Please provide a comma separated list without blanks (eg.: group1,group2). Default: empty

**Group Quotas**: Define quotas for groups of the users authenticated via CAS. Please provide a comma separated list without blanks and with : between group names and quotas (eg.: group1:10GB,group2:500MB). Default: empty

**Group Name Filter**: Define a filter (PHP RegExp syntax!) with only the allowed characters for a group name. Group names are cut after 63 characters per definition by ownCloud/Nextcloud core and appended by an horizontal ellipsis. Default when empty: `a-zA-Z0-9\.\-_ @`

**Group Name Replace Umlauts**: Activate to filter german umlauts out of the group’s name. Only works, if *Group* in "Mapping" is filled. Default: off



ECAS Settings:
--------------

Since Version 1.5 user_cas provides support for using a European Commission ECAS-Server implementation.

**Use ECAS Attribute Parser?**: Activate the ECAS attribute parser to enable the parsing of groups provided by the European Commission ECAS implementation (do **NOT** activate until you know what you are doing).

**Request full user details?**: Activate to request a full user profile in the ECAS callback (do **NOT** activate until you know what you are doing).

**ECAS Strength**: Set the authentication strength used by the ECAS instance when validating a user’s ticket (do **NOT** select until you know what you are doing).

**ECAS AssuranceLevel**: Set the assurance level used by the ECAS instance when validating a user’s ticket (do **NOT** select until you know what you are doing).

**Query ECAS groups**: Define which ECAS groups should be queried when validating a user’s ticket. Please provide a comma separated list without blanks (eg.: GROUP1,GROUP2 or use * for all groups). (Do **NOT** select until you know what you are doing).

**Don’t use Multi-Factor-Authentication on these client-IPs**: Comma separated list of client IP addresses (or address ranges), which won’t be forced to use Multi-Factor-Authentication if "ECAS AssuranceLevel" is at least MEDIUM (e.g. 192.168.1.1-254,192.168.2.5). (Do **NOT** fill until you know what you are doing).


Import CLI:
-----------

Since Version Version 1.7.2 user_cas provides support for importing users from an ActiveDirectory LDAP.

#### ActiveDirectory (LDAP): Provide the necessary information to connect to your AD LDAP Server

**LDAP-Host**: Provide the LDAP-host information. Set the protocol, host and port to use. Default: empty

**LDAP-User and Domain**: Provide the LDAP user and domain to authenticate the LDAP connection Default: empty

**LDAP-User Password**: Set the password for the user (see above). Default: empty

**LDAP Base DN**: Set the LDAP Base Distinguished Name (DN) for the query. Default: empty

**LDAP Sync Filter**: Define the filter to be used when querying the according user’s from LDAP. Default: empty

**LDAP Syn Pagesize (1-1500)**: Define the pagesize of the LDAP query response according to your LDAP server’s settings. Default: 1500

#### CLI Attribute Mapping: Provide the necessary information to map your AD LDAP users to ownCloud

**UID/Username**:  Name of uid/username attribute in your LDAP response. Default: empty

**Display Name**:  Name of display name attribute(s) in your LDAP response (this might be the "real name" of a user or a combination of two fields like: givenname+sn). Default: empty

**Email**: Name of email attribute in your LDAP response. Default: empty

**Group**: Name of group attribute in your LDAP response. Default: empty

**Group Name Field**: Name of the LDAP attribute in your group node to set a group’s name. If no name filed is set or found, the DN of the group will be used as the group’s name. Default: empty

**Quota**: Name of quota attribute in your LDAP response. Quota can be a numeric byte value or a human readable string, like 1GB or 512MB. Default: empty

**Enable**: Name of the enable attribute in your LDAP response. This sets the account to enabled/disabled on import, while enabled = 1 and disabled = 0. Default: empty

**Calculate Enable Attribute Bitwise AND with**: Provide to use a bitwise AND calculation to define the enabled status of the account. Only use if your LDAP’s enabled attribute value is not 0|1. Default: empty

**Merge Accounts**: Activate to enable account merging. Default: off

**Prefer Enabled over Disabled Accounts on Merge**: Activate to prefer enabled second accounts over disabled primary accounts. Only works, if *Merge Accounts* is enabled. Default: off

**Merge two active accounts by**: Name of the attribute in your LDAP response by which you want to merge two activated accounts. Only works, if *Merge Accounts* is enabled. Default: empty

**Merge two active accounts by: Filterstring**: Define a filterstring for the *Merge by* attribute, that defines which activated account should be preferred on merge. Only works, if *Merge by* is set and *Merge Accounts* is enabled. Default: empty



<!-- 
LDAP-Backend:
-------------

**Link to LDAP backend**: Link CAS authentication with LDAP users and groups backend to use the same ownCloud user as if the user was logged in via LDAP. 

-->


PHP-CAS Library
---------------

Setting up the PHP-CAS library options :

**Overwrite phpCAS path (CAS.php file)**: Set a custom path to a CAS.php file of the jasig/phpcas library version you want to use. Beginning at rootlevel of your server. Default: empty, meaning it uses the composer installed dependency in the `user_cas` folder.

**PHP CAS debug file**: Set path to a custom phpcas debug file. Beginning at rootlevel of your server. Default: empty


EXTRA INFO
==========

* If you enable the "Autocreate user after CAS login" option, a user will be created if he does not exist. If this option is disabled and the user does not exist, then the user will be not allowed to log into ownCloud. <!-- You might not want this if you check "Link to LDAP backend" -->

* If you enable the "Update user data" option, the app updates the user's Display Name, Email, Group Membership and Quota on each login.

By default the CAS App will unlink all the groups from a user and will provide the groups defined at the [**Mapping**](#mapping) attributes. If this mapping is not defined, the value of the *Default group* field will be used instead. If both are undefined, then the user will be set with no groups.
If you set the *Locked Groups* field, those groups will not be unlinked from the user.


OCC Commands
============

user_cas has the following OCC commands implemented:

* cas
    * cas:create-user (Adds a user_cas user to the database.)
    * cas:update-user (Updates an existing user and, if not yet a CAS user, converts the record to CAS backend.)
    * cas:import-users-ad (Imports users from an ActiveDirectory LDAP.)


Create a user:
--------------

    cas:create-user [--display-name [DISPLAY-NAME]] [--email [EMAIL]] [-g|--group [GROUP]] [-o|--quota [QUOTA]] [-e|--enabled [ENABLED]] [--] <uid>
    
- Parameters (required):
    - uid: the uid of the user
    
- Options (optional):
    - --display-name: The new display name of the user.
    - --email: The new email of the user.
    - -g | --group: The new group of the user, can be used multiple times (e.g. `-g Family -g Work`) to add multiple groups.
    - -o | --quota: The new quota of the user, either as numerical byte value or human readable value (e.g. 1GB)).
    - -e | --enabled: Enable or disable the user. Setting `-e 1` enables the user, setting `-e 0` disables the user.
    
**Notice: Protected groups will never be unlinked from the user! See also [Groups](#groups).**
    

Update a user:
--------------

    cas:update-user [--display-name [DISPLAY-NAME]] [--email [EMAIL]] [-g|--group [GROUP]] [-o|--quota [QUOTA]] [-e|--enabled [ENABLED]] [-c|--convert-backend [CONVERT-BACKEND]] [--] <uid>

- Parameters (required):
    - uid: the uid of the user
    
- Options (optional):
    - --display-name: The new display name of the user.
    - --email: The new email of the user.
    - -g | --group: The new group of the user, can be used multiple times (e.g. `-g Family -g Work`) to add multiple groups.
    - -o | --quota: The new quota of the user, either as numerical byte value or human readable value (e.g. 1GB)).
    - -e | --enabled: Enable or disable the user. Setting `-e 1` enables the user, setting `-e 0` disables the user.
    - -c | --convert-backend: Set if the user’s backend should be converted to CAS backend. Setting `-c 1` converts to backend to CAS. **WARNING: This is not revocable!**
    
**Notice: Protected groups will never be unlinked from the user! See also [Groups](#groups).**


Import users from ActiveDirectory (LDAP):
-----------------------------------------

    cas:import-users-ad [-d|--delta-update [1]] [-c|--convert-backend [1]]

- Options (optional):
    - -d | --delta-update: Enable or disable delta updates of accounts. Setting `-d 1` enables account updates.
    - -c | --convert-backend: Set if the user’s backend should be converted to CAS backend. Setting `-c 1` converts to backend to CAS. **WARNING: This is not revocable!**


**Additional Info:** If you want to automate the ActiveDirectory import, call this command in a cronjob of your webservers user (e.g. `www-data` on debian based linux systems).

**Additional Info:** If you want additional debug information, use the options `-vv` or `-vvv` in your command, to raise the debug level. To quiet the output, use the option `-q`

**Example for usage as daily cronjob with delta updates and backend conversion (if needed), output will be zeroed**: 

    0 0 * * * /usr/bin/php /path/to/owncloud/occ cas:import-users-ad -d 1 -c 1 -q >/dev/null 2>&1


Bugs and Support
==============

Please contribute bug reports and feedback to [GitHub Issues](https://github.com/felixrupp/user_cas/issues).  

Commercial support and feature implementation is available via [felixrupp.com](http://www.felixrupp.com).


ABOUT
=====

License
-------

AGPL 3.0 or later - http://www.gnu.org/licenses/agpl-3.0.html

Authors
-------

Current Version, since 1.4.0:
* Felix Rupp - [github.com/felixrupp](https://github.com/felixrupp)

Older Versions:
* Sixto Martin Garcia - [github.com/pitbulk](https://github.com/pitbulk)
* David Willinger (Leonis Holding) - [github.com/leoniswebDAVe](https://github.com/leoniswebDAVe)
* Florian Hintermeier (Leonis Holding)  - [github.com/leonisCacheFlo](https://github.com/leonisCacheFlo)
* brenard - [github.com/brenard](https://github.com/brenard)

Links
-------
* ownCloud - [owncloud.org](http://www.owncloud.org)
* ownCloud @ GitHub - [github.com/owncloud](https://github.com/owncloud)

CHANGELOG.md
CHANGELOG
=========

Version 1.8.4
-------------

* Adds option to disable SingleSignout requests to the ownCloud/Nextcloud instance in admin panel
* Adds missing translations for options and readme entries


Version 1.8.3
-------------

* Fixes SingleSignOut feature - thanks to [@pingou2712](https://github.com/pingou2712>)
* Fixes wrong link to Nextcloud AppStore in Readme - thanks to [@Emi94](https://github.com/Emi94)


Version 1.8.2
-------------

* Raises compatibility to ownCloud 10.4
* Adds section in README to describe how to disable code integrity checks on dev setups


Version 1.8.1
-------------

* Raises compatibility to Nextcloud 18

* Fixes Nextcloud 18 WebDAV bug - thanks to [@landryb](https://github.com/landryb)
* Fixes internal ip checks for *»Don’t use Multi-Factor-Authentication on these client-IPs«*
* Fixes typo in acceptedStrength \[ECAS\]


Version 1.8.0
-------------

* Drops compatibility for Nextcloud 13
* Raises compatibility to Nextcloud 17

* Fixes a redirect error with enabled enforce-authentication and non-authorized users
* Fixes a security bug, where a user, not authorized by any groups, still had the possibility to gain access to a read only account
* Fixes deprecated quota handling: all quotas should now be handled correctly (even 'none' and 'default')

* Adds option to overwrite the CAS login button’s label
* Adds option to keep CAS-ticket-ids in URL (see [Issue #17](https://github.com/felixrupp/user_cas/issues/17))
* Adds feature to use CAS Single-Logout 

* Bumps phpCAS library to 1.3.8


Version 1.7.4
-------------

* Fixes CAS logout service backlink


Version 1.7.3
-------------

* Raises compatibility to ownCloud 10.3
* Adds support for cas server path "/" - thanks to [@dattn](https://github.com/dattn)


Version 1.7.2
-------------

* Fixes double synchronisation of DisplayName, causing overlays in DisplayName on first login
* Fixes problems with local database users being converted to user_cas backend by accident
* Fixes logout handling to not need a list of Logout Servers anymore

* Adds error handling when CAS basic config is missing or truncated
* Adds the license file for AGPL 3.0 or later
* Adds ActiveDirectory import OCC command, let’s you import specific LDAP users as user_cas users to your ownCloud/Nextcloud instance
* Adds options for group name filter and group umlaut replacement in "Groups" panel


Version 1.7.1
-------------

* Raises compatibility to ownCloud 10.2 and Nextcloud 16
* Fixes Create-/UpdateCommand and synchronisation with the user_cas-Backend
* Fixes user autocreation process and synchronisation with the user_cas-Backend
* Fixes Nextcloud detection algorithm, it does not depend on a user-editable string anymore


Version 1.7.0
-------------

* Adds OCC commands to create and update CAS users directly via command line
* Adds user quota mapping input in Settings Panel - user quota overwrites group quotas
* Adds a filter to disable multi-factor-authentication for internal IP addresses on ECAS instances
* Raises the ECAS client-version to 2.1.3
* Fixes postLogin signals signature


Version 1.6.2
-------------
* Changes default value of the autocreate option to "on"
* Fixes postLogin signals signature


Version 1.6.1
-------------
* Fixes 5min logout bug on Nextcloud
* Unifies log levels in LoggingService
* Moves hook registration to a higher level to ensure working postLoginHook


Version 1.6.0
-------------
* Adds compatibility for ownCloud 10.1 and Nextcloud 15
* Fixes app initialization to minimize incompatibilities with other apps
* Removes hard app stopping when no phpCAS library is found
* Updates jasig/phpcas to 1.3.6


Version 1.5.6
-------------
* Changes most of the INFO log writes to DEBUG


Version 1.5.5
-------------
* Adds Support for ownCloud 10.0.10 and Nextcloud 14.0.0


Version 1.5.4
-------------
* Fixes Nextcloud log flood bug
* Fixes documentation links in info.xml
* Fixes ownCloud/Nextcloud Session Token-Password bug
* Fixes enforce authentication issues with Database Backend
* Fixes enforce authentication issues with public gallery interface


* Adds support for ownCloud until version 10.0.9 and Nextcloud until version 13.0.6
* Adds documentation for basic installation via release archive


* Removes hardcoded PHP version requirements


Version 1.5.3
-------------
* Hotfixes the IP address range separator in the exclude specific IPs field and changes it from "/" to "-"
* Fixes ownCloud 10.0.8 bug and raises compatibility to 10.0.8

Version 1.5.2
-------------
* Add settings field to exclude specific Ips and/or IP-ranges from force login
* Nextcloud: Move settings panel to section "Security" (was in "Additional" before)
* Remove the signature from repo (if you need a signed version, please use one of the release packages or download from ownCloud Market/Nextcloud AppStore)

Version 1.5.1
-------------
* Hotfixes wrong links in 403 error page if enforce authentication was on
* Hotfixes wrong translation in 403 error page for ECAS instances
* Removes return type hints not compatible with PHP 5.6
* Adds functionality to provide more than one mapping field to ownCloud userdata fields (e.g. DisplayName can now be concatenated by a firstname and a lastname CAS-field)

Version 1.5.0
-------------
* Drop ownCloud 9 support
* Major source code optimizations, fix several errors associated with redirection after login
* Add ECAS support
* Add authorization feature via groups
* Add error views for when not authorized or when the CAS-Client throws errors
* Support for ownCloud oauth2 App, it’s now possible to authenticate a desktop or mobile phone client with CAS via oauth2 provider

Version 1.4.9
-------------
* Hotfixes the autocreate bug, mentioned in Issue [#13](https://github.com/felixrupp/user_cas/issues/13).

Version 1.4.8
-------------
* Hotfixes the current 1.4 version to fix a major bug preventing the OCS-Api to work, while the user_cas app is installed and enabled.

Version 1.4.7
-------------

l10n/de.js
OC.L10N.register(
    "user_cas",
    {
        "CAS Authentication backend": "CAS Authentifizierungsbackend",
        "CAS Server": "CAS Server",
        "Basic": "Grundeinstellungen",
        "Mapping": "Zuordnungen",
        "phpCAS Library": "phpCAS Bibliothek",
        "CAS Server Version": "CAS Server Version",
        "CAS Server Hostname": "CAS Server Hostname",
        "CAS Server Port": "CAS Server Port",
        "CAS Server Path": "CAS Server Pfad",
        "Service URL": "Dienst URL",
        "Certification file path (.crt).": "Pfad zur Zertifikatsdatei (.crt).",
        "Leave empty if you don’t want to validate your CAS server instance": "Nicht ausfüllen, wenn Sie ihren CAS-Server nicht validieren möchten",
        "Use CAS proxy initialization": "CAS Proxy Initialisierung verwenden",
        "Force user login using CAS?": "CAS Login erzwingen?",
        "Don’t use force login on these client-IPs": "Login bei diesen Client-IPs nicht erzwingen",
        "Comma separated list of client IP addresses (or address ranges), which won’t be forced to login if \"Force user login\" is enabled (e.g. 192.168.1.1-254,192.168.2.5)": "Kommagetrennte Liste an Client-IP-Adressen (oder Adressräumen), die nicht zum Login gezwungen werden, wenn \"CAS Login erzwingen\" aktiviert ist (z.B: 192.168.1.1-254,192.168.2.5)",
        "Disable CAS logout (do not logout CAS-session, only instance-session)": "CAS Logout deaktivieren (nicht die CAS-Session, sondern nur die Instanz-Session ausloggen)",
        "Disable CAS SingleSignout (do not logout instance-session if CAS-server sends SSO-Request)": "CAS SingleSignout deaktivieren (nicht die Instanz-Session ausloggen, wenn der CAS-Server einen SSO-Request sendet)",
        "SingleSignout Servers": "Logout Server",
        "Comma separated list of servers which can send SingleSignout requests (leave empty if you do not have to restrict SingleSignout to defined servers)": "Kommagetrennte Liste an Servern, die SingleSingout-Requests an ihren CAS-Server senden dürfen (Nicht ausfüllen, wenn Sie SingleSignouts nicht beschränken/erlauben müssen)",
        "Autocreate user after first CAS login?": "Benutzer nach erstem CAS Login automatisch erstellen?",
        "Keep CAS-ticket-ids in URL?": "CAS-Ticket-IDs nicht aus der URL entfernen",
        "Beware: Potential security risk! Only activate, if you know what you are doing.": "Achtung: Potentielles Sicherheitsrisiko! Nur aktivieren, wenn Sie wissen, was Sie tun.",
        "Overwrite Login Button Label": "Login Button Label überschreiben",
        "Link CAS authentication with LDAP users and groups backend": "CAS Authentifizierung mit LDAP Backend verknüpfen",
        "Update user data after each CAS login?": "Benutzerdaten nach jedem CAS Login aktualisieren?",
        "Locked Groups": "Geschützte Gruppen",
        "Groups that will not be unlinked from the user when sync the CAS server and the owncloud": "Gruppen, die beim Synchronisieren mit dem CAS Server nicht vom ownCloud Benutzer entfernt werden",
        "Multivalued field, use comma to separate values": "Mehrzeiliges Feld, nutzen Sie komma separierte Werte",
        "Default Group": "Standardgruppe",
        "Default group when autocreating users and no group data was found for the user": "Standardgruppe, die beim automatischen Neuerstellen von ownCloud Benutzern vergeben wird",
        "Authorized CAS Groups": "Authorisierte CAS Gruppen",
        "Users in the following groups will be able to log into ownCloud, users not in one of the groups will be logged out immediately": "Benutzer in den folgenden Gruppen können sich in ownCloud anmelden, Benutzer, die nicht in einer der Gruppen sind, werden sofort ausgeloggt",
        "Group Quotas": "Gruppenquotas",
        "Use ECAS Attribute Parser?": "ECAS Attribute Parser verwenden?",
        "Request full user details?": "Vollständige Benutzerdetails anfordern?",
        "ECAS Strength": "ECAS Strength",
        "Not set": "Nicht gesetzt",
        "ECAS AssuranceLevel": "ECAS AssuranceLevel",
        "Query ECAS groups": "Gruppen in ECAS-Abfrage",
        "Note down all groups which you want to receive from your ECAS instance, * returns all groups": "Setzen Sie alle Gruppen, die Sie von Ihrer ECAS-Instanz zurück erhalten möchten, * gibt alle Gruppen zurück",
        "Don’t use Multi-Factor-Authentication on these client-IPs": "Multi-Faktor-Authentifizierung bei diesen Client-IPs deaktivieren",
        "Comma separated list of client IP addresses (or address ranges), which won’t be forced to use Multi-Factor-Authentication if \"ECAS AssuranceLevel\" is at least MEDIUM (e.g. 192.168.1.1-254,192.168.2.5)": "Kommagetrennte Liste an Client-IP-Adressen (oder Adressräumen), die nicht zur Multi-Faktor-Authentifizierung gezwungen werden, wenn der \"ECAS AssuranceLevel\" mindestens MEDIUM ist (z.B: 192.168.1.1-254,192.168.2.5)",
        "Email": "E-Mail",
        "Display Name": "Anzeigename",
        "Groups": "Gruppen",
        "Overwrite phpCAS path (CAS.php file)": "phpCAS Pfad überschreiben (CAS.php Datei)",
        "Optional: Overwrite phpCAS path (CAS.php file) if you want to use your own version. Leave blank to use the shipped version.": "Optional: Überschreiben Sie den phpCAS Pfad (CAS.php Datei), wenn Sie eine eigene Version verwenden möchten. Nicht ausfüllen, wenn Sie die Standardversion verwenden möchten",
        "PHP CAS debug file": "PHP CAS debug Datei",
        "Save": "Speichern",
        "Your CAS settings have been updated.": "Ihre CAS Einstellungen wurden gespeichert.",
        "Your CAS settings could not be updated. Please try again.": "Ihre CAS Einstellungen konnten nicht gespeichert werden. Bitte versuchen Sie es erneut.",
        "Forbidden. You do not have access to this application. Please refer to your administrator if something feels wrong to you.": "Verboten. Sie haben keinen Zugriff auf diese Anwendung. Bitte wenden Sie sich an Ihren Administrator, wenn Sie denken, dass etwas nicht stimmt.",
        "You do not have access to the JRCbox application. Please contact the JRCbox administrator if something feels wrong to you.": "Sie haben keinen Zugriff auf diese JRCbox Anwendung. Bitte wenden Sie sich an den JRCbox Administrator, wenn Sie denken, dass etwas nicht stimmt.",
        "Internal Server Error. The server encountered an error. Please try again.": "Interner Serverfehler. Die Serveranfrage war fehlerhaft, bitte versuchen Sie es erneut.",
        "Go back to the login page": "Zurück zur Startseite",

        "Import CLI": "Import CLI",
        "ActiveDirectory (LDAP)": "ActiveDirectory (LDAP)",
        "LDAP Host": "LDAP Host",
        "LDAP User and Domain": "LDAP Benutzer und Domain",
        "LDAP User Password": "LDAP Benutzerpasswort",
        "LDAP Base DN": "LDAP Basis DN",
        "LDAP Sync Filter": "LDAP Sync Filter",
        "LDAP Sync Pagesize (1–1500)": "LDAP Sync Seitengröße (1–1500)",
        "CLI Attribute Mapping": "CLI Attributzuordnung",
        "UID/Username": "UID/Benutzername",
        "Group Name Field": "Gruppen Namensfeld",
        "Group Name Filter": "Gruppen Namensfeld Filter",
        "Attention: You must use PHP (PCRE) Regex syntax for the filter.": "Achtung: Sie müssen die PHP (PCRE) Regex Syntax für diesen Filter einhalten.",
        "Group Name Replace Umlauts": "Gruppen Name: Ersetze Umlaute",
        "Enable": "Aktivierung",
        "Calculate Enable Attribute Bitwise AND with": "Berechne die Nutzer-Aktivierung 'Bitwise AND' mit",
        "Merge Accounts": "Accounts zusammenführen",
        "Prefer Enabled over Disabled Accounts on Merge": "Aktivierte Accounts beim Zusammenführen bevorzugen",
        "Merge Two Active Accounts by": "Zwei aktivierte Accounts zusammenfügen mit Hilfe vom Attribut",
        "Merge Two Active Accounts by: Filterstring": "Filterstring für das Zusammenfügen zweier aktivierter Accounts"
    },
    "nplurals=2; plural=(n != 1);");
